src/Analyzer/Pass/Expression/ArrayDuplicateKeys.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @author Kévin Gomez https://github.com/K-Phoen <contact@kevingomez.fr>
+ */
+
+namespace PHPSA\Analyzer\Pass\Expression;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Scalar;
+use PHPSA\Analyzer\Pass;
+use PHPSA\Context;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+class ArrayDuplicateKeys implements Pass\AnalyzerPassInterface, Pass\ConfigurablePassInterface
+{
+    /**
+     * @param Expr\Array_ $expr
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(Expr\Array_ $expr, Context $context)
+    {
+        $result = false;
+        $keys = [];
+
+        /** @var Expr\ArrayItem $item */
+        foreach ($expr->items as $item) {
+            if (!$item->key instanceof Scalar) {
```

Comment:
Maybe will be better to use Expression Compiler to support it? and working with CompiledExpression
instead of working `Scalar` Node types
What do you think?